server/plugins/OscUGens.cpp

```diff
@@ -2031,7 +2025,14 @@ void VOsc_Ctor(VOsc *unit)
 
 	unit->m_phasein = ZIN0(2);
 	unit->m_phaseoffset = (int32)(unit->m_phasein * unit->m_radtoinc);
-	unit->m_phase = unit->m_phaseoffset;
+
+	if (INRATE(2) == calc_FullRate) {
+		SETCALC(VOsc_next_ika);
+		unit->m_phase = 0;
```

Comment:
The original post has a test case for this -- if I didn't set the phase to 0, it caused an extra initial phase offset.

Exactly why, I'm not totally sure. But SinOsc does the same thing, setting the initial phase to 0 for all cases except for when both the freq and phase arguments are control-rate: https://github.com/supercollider/supercollider/blob/master/server/plugins/OscUGens.cpp#L1540